near-workspaces/__tests__/market.ava.ts

```diff
@@ -0,0 +1,151 @@
+    const john = await root.createAccount('john')
+
+    const contract = await root.createAndDeploy(
+        'cnft',
+        '../build/release/cNFT.wasm',
+        {
+            method: 'init',
+            args: {
+                owner_id: alice.accountId,
+                contract_metadata: CONTRACT_METADATA,
+                contract_extra: {
+                    ...CONTRACT_EXTRA,
+                    ...{ mints_per_address: 2, max_copies: 3 },
+                },
+            },
+        }
+    )
+
+    log(`✓  Alice initiate the contract successfully\n`)
+
+    return { alice, john, contract }
+})
+
+workspace.test(
+    'Should bid one a token then fetch it with the right data',
+    async (test, { contract, alice, john }) => {
+        const { result: minted } = await call_mint(contract, alice)
+
+        const alice_example_bid = {
+            amount: toYocto(randomInt(0, 100).toString()),
+            bidder: alice.accountId,
+            recipient: minted.id,
+            sell_on_share: randomInt(0, 20),
+            currency: 'near',
+        }
+        const john_example_bid = {
+            amount: '10000000000000000000000',
+            bidder: john.accountId,
+            recipient: minted.id,
+            sell_on_share: randomInt(0, 20),
+            currency: 'near',
+        }
+
+        /**
+         *  @todo fix in contract
+         *  user shouldn't be able to bid on his own token
+         * */
+        // await test.throwsAsync(async () => {
+        //     await call_set_bid(contract, alice, {
+        //         tokenId: minted.id,
+        //         bid: alice_example_bid
+        //     })
+        // })
+        // log(`✓  Alice failed to bid on her own token\n`)
+
+        /**
+         *  @todo fix in contract
+         *  user shouldn't be able to create bid for others
+         * */
+        // await test.throwsAsync(async () => {
+        //     await call_set_bid(contract, john, {
+        //         tokenId: minted.id,
+        //         bid: alice_example_bid
+        //     })
+        // })
+        // log(`✓  John failed to bid using alice account\n`)
+
+        await test.throwsAsync(async () => {
+            await call_set_bid(contract, john, {
+                tokenId: '6807',
+                bid: john_example_bid,
+            })
+        })
+        log(`✓  John failed to bid on non-existant token\n`)
+
+        await test.throwsAsync(async () => {
+            await call_set_bid(contract, john, {
+                tokenId: minted.id,
+                bid: { ...john_example_bid, amount: 0 },
+            })
+        })
+        log(`✓  John failed to bid 0\n`)
+
+        const johnBalanceBefore = await john.availableBalance()
+        const contractBalanceBefore = await contract.availableBalance()
+
+        await call_set_bid(contract, john, {
+            tokenId: minted.id,
+            bid: {
+                amount: '10000000000000000000000',
```

Comment:
Does `NEAR.parse("1N")` work here? It would be better for readability and to avoid amount typo.

https://github.com/near/workspaces-js/blob/512dd2457bbec81a5fde751cb05c086e0ac7ffbc/__tests__/05.spoon-contract-to-sandbox.ava.ts#L194